compiler/semantics/ir_visitor.py

```diff
@@ -757,17 +817,18 @@ def visitDispense(self, ctx: BSParser.DispenseContext):
             size = deff['index']
 
         # Grab the declared volume of the variable and store it
-        _volume = 0;
-        if (type(ctx.unitTracker()) != BSParser.UnitTrackerContext):
-            _volume = 10; # Default to 10 if the volume hasn't been explicitly declared
+        _volume = 0
+        if type(ctx.unitTracker()) != BSParser.UnitTrackerContext:
+            _volume = 10  # Default to 10 if the volume hasn't been explicitly declared
         else:
-            _volume = int(ctx.unitTracker().INTEGER_LITERAL().__str__())
-
-        #print("Volume: " + str(_volume)) #DEBUG
+            _volume = deepcopy(int(ctx.unitTracker().INTEGER_LITERAL().__str__()))
 
         # We don't have to check here, because this is a dispense.
 
-        #print("Scope_Stack: " + str(self.scope_stack[-1]))
+        if deff['name'] in self.symbol_table.get_local(deff['name'], self.scope_stack[-1]).volumes:
+            self.symbol_table.get_local(deff['name'], self.scope_stack[-1]).volumes[deff['name']].append(_volume)
```

Comment:
Here you're using the variable name as the key to the volumes dictionary, which requires you to use some hacky manipulation to retrieve values in the volume tracker -- the whole point of using a dictionary is for uniqueness.  This works _for tested cases_, but as you are using the index of the list to retrieve the value based on the _assumed name_ from SSA-renaming, this does not generalize.  In short, the dict key should be the instruction, not the name, and when you retrieve a value in volume tracker you'll need that same line number.

You can get the _unique_ instruction id (`iid`) from the ir that is generated a few lines after this.

You will want to do the same for the mix value you parse here (unique dict items keyed by the `iid`), as there is nothing stopping us from doing:

```
a = dispense ....
a = mix x with y
```
Which would overwrite the volume attribute in the symbol table currently.